codex/sales/states/unknown.nim

```diff
@@ -0,0 +1,47 @@
+import ../statemachine
+import ./filled
+import ./finished
+import ./failed
+import ./errored
+import ./cancelled
+
+type
+  SaleUnknown* = ref object of SaleState
+  SaleUnknownError* = object of CatchableError
+
+method `$`*(state: SaleUnknown): string = "SaleUnknown"
+
+method onCancelled*(state: SaleUnknown, request: StorageRequest) {.async.} =
+  await state.switchAsync(SaleCancelled())
+
+method onFailed*(state: SaleUnknown, request: StorageRequest) {.async.} =
+  await state.switchAsync(SaleFailed())
+
+method enterAsync(state: SaleUnknown) {.async.} =
+  without agent =? (state.context as SalesAgent):
+    raiseAssert "invalid state"
+
+  let market = agent.sales.market
+
+  try:
+    without requestState =? await market.requestState(agent.requestId):
+      raiseAssert "state unknown"
```

Comment:
This would kill the node when the state cannot be retrieved from the contract. Seems a bit harsh :sweat_smile: